### PR TITLE
Stepangranat/dashboard and history

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

File: webapp/src/component/activity/types/getTextDiff.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const newInput = input?.new;
+  if (oldInput && newInput) {
+    const diffed = diffWordsWithSpace(oldInput, newInput);
+    return (
+      <StyledDiff>
+        {diffed.map((part, i) => {
+          return (
+            <span
+              key={i}
+              className={clsx({ added: part.added, removed: part.removed })}
```
Comment (JanCizmar) [inline webapp/src/component/activity/types/getTextDiff.tsx:28]: Here it's also not clear, that the span may have only one of the classes.

File: webapp/src/views/projects/dashboard/ProjectTotals.tsx [hunk 2]
```diff
@@ -121,41 +267,81 @@ export const ProjectTotals: FC<{
...
+          )}
+        </StyledTile>
 
-        <StyledTile>
+        <StyledTile
+          gridArea="tags"
+          className={clsx({ clickable: tagsPresent })}
+          onClick={tagsPresent ? handleMenuOpen : undefined}
+          aria-controls={open ? 'basic-menu' : undefined}
+          aria-haspopup="true"
```
Comment (JanCizmar) [inline webapp/src/views/projects/dashboard/ProjectTotals.tsx:312]: Accessibility? That's what we're doing now?

**Review by JanCizmar — APPROVED** [review]

File: webapp/src/component/activity/types/getCommentStateChange.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      <StyledSuccess className={type}>
+        <Check fontSize="small" />
+        <div>
+          <T keyName="translations_comments_resolved" />
+        </div>
+      </StyledSuccess>
+    );
+  } else {
+    return (
+      <StyledProblem className={type}>
```
[the remark this answers is not in the document]
Comment (stepan662) [reply webapp/src/component/activity/types/getCommentStateChange.tsx:46]: 👍

File: webapp/src/component/activity/types/getProjectLanguageChange.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import { DiffValue } from '../types';
+
+const StyledCircledLanguageIcon = styled(CircledLanguageIcon)`
+  display: inline-block;
+  vertical-align: text-bottom;
+`;
+
+const StyledDiff = styled('span')`
+  word-break: break-word;
+  & .removed {
```
[the remark this answers is not in the document]
Comment (stepan662) [reply webapp/src/component/activity/types/getProjectLanguageChange.tsx:12]: I've removed it where it's not necessary

File: webapp/src/component/activity/types/getTextDiff.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { styled } from '@mui/material';
+import clsx from 'clsx';
+import { diffWordsWithSpace } from 'diff';
+
+import { DiffValue } from '../types';
+
+const StyledDiff = styled('span')`
+  & .removed {
```
[the remark this answers is not in the document]
Comment (stepan662) [reply webapp/src/component/activity/types/getTextDiff.tsx:8]: 👍

File: webapp/src/component/activity/types/getTextDiff.tsx (hunk 1 above)
Comment (stepan662) [reply webapp/src/component/activity/types/getTextDiff.tsx:28]: 👍

File: webapp/src/component/activity/configuration.tsx [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        type: 'text',
+      },
+      state: {
+        type: 'comment_state',
+      },
+    },
+    references: ({ relations, description }) => {
+      const result: Reference[] = [];
+      // @ts-ignore
+      const translationRelations = relations?.translation?.relations;
```
[the remark this answers is not in the document]
Comment (JanCizmar) [reply webapp/src/component/activity/configuration.tsx:125]: (relations?.translation as any).relations ?

**Pull request merged**